crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -40,8 +40,24 @@ def role_constraints
   # Unfortunatelly we need to explicitely look at crowbar-status of the proposal
   # because apply_role from this model ignores errors from superclass's apply_role.
   def commit_and_check_proposal
-    proposal_commit("default", false, false)
+    answer = proposal_commit("default", false, false)
+    found_errors = []
+    # check if error message is saved in one of the nodes
+    if answer.first != 200
+      NodeObject.find("state:crowbar_upgrade").each do |node|
+        error = node["crowbar_wall"]["chef-error"] || ""
+        next if error.empty?
+        found_errors.push error
+        @logger.error("Chef run ended with an error on the node #{name}: #{error}")
+        node["crowbar_wall"]["chef-error"] = ""
+        node.save
+      end
+    end
+    unless found_errors.empty?
+      raise found_errors.join("\n") # this will not work for flash messages...
+    end
```

Comment:
Does it make sense to move all the use of  `found_errors` inside the `if` block?